### PR TITLE
feat: add useClearTextShortcut setting

### PR DESCRIPTION
**Pull request by ppken**
Addresses: https://github.com/appium/appium/issues/20727

Added `useClearTextShortcut` setting to be able to enable / disable text clear by HIDEvent.

### CODE REVIEW

File: WebDriverAgentLib/Commands/FBSessionCommands.m [hunk 1]
```diff
@@ -141,6 +141,10 @@ + (NSArray *)routes
     [FBConfiguration forceSimulatorSoftwareKeyboardPresence];
   }
 
+  if (capabilities[FB_CAP_USE_HID_CLEAR]) {
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Commands/FBSessionCommands.m]: it is not really needed to add a capability for it. Settings could also be set upon session startup

File: WebDriverAgentLib/Utilities/FBSettings.m [hunk 2]
```diff
@@ -35,3 +35,4 @@
 NSString* const FB_SETTING_ANIMATION_COOL_OFF_TIMEOUT = @"animationCoolOffTimeout";
 NSString* const FB_SETTING_MAX_TYPING_FREQUENCY = @"maxTypingFrequency";
 NSString* const FB_SETTING_RESPECT_SYSTEM_ALERTS = @"respectSystemAlerts";
+NSString* const FB_SETTING_USE_HID_CLEAR = @"useHIDClear";
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Utilities/FBSettings.m]: how about changing the name to `useClearTextShortcut` ?

File: WebDriverAgentLib/Commands/FBSessionCommands.m (hunk 1 above)
Comment (ppken) [reply WebDriverAgentLib/Commands/FBSessionCommands.m]: Thank you! removed the capability in https://github.com/appium/WebDriverAgent/pull/952/commits/35afc8ffbbeae9fcd098a1ae292c18ec9b0ff52f and fixed to use setting instead in https://github.com/appium/WebDriverAgent/pull/952/commits/e89b0f77a76ac2ee56826c79bc7ac09cad99781c

File: WebDriverAgentLib/Utilities/FBSettings.m (hunk 2 above)
Comment (ppken) [reply WebDriverAgentLib/Utilities/FBSettings.m]: Thank you! fixed in https://github.com/appium/WebDriverAgent/pull/952/commits/e89b0f77a76ac2ee56826c79bc7ac09cad99781c

**Comment by mykola-mokhnach** [comment]
@ppken please do not forget to document the new setting in https://github.com/appium/appium-xcuitest-driver/blob/master/docs/reference/settings.md and bump the minimum WDA version there after this change is published

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by ppken** [comment]
@mykola-mokhnach 

I'm so sorry, but maybe I misunderstood your suggestion and I added these lines to set `useClearTextShortcut` via capability.

https://github.com/ppken/WebDriverAgent/blob/b639620a70fb9a489fd728f64af85bccf6c3b36d/WebDriverAgentLib/Commands/FBSessionCommands.m#L144-L146

May I remove these lines in another PR?

**Comment by jlipps** [comment]
Hi @ppken, congrats: the Appium project wants to compensate you for this contribution! Please reply to this comment mentioning @jlipps and @KazuCocoa and sharing your [OpenCollective](https://opencollective.com/) account name, so that we can initiate payment! Or let us know if you decline to receive compensation via OpenCollective. Thank you!

**Comment by ppken** [comment]
@jlipps @KazuCocoa
Thank you for your offer!

This contribution was a part of our company's work, MagicPod.
So, I decided to decline your offer for this time as an employee of MagicPod.

**Pull request merged**